### PR TITLE
chore(tools): Fix doc inclusion tools on Windows

### LINKED ISSUE #13681: `make docs` doesn't work on Windows
Running `make docs` on Windows doesn't work.

### Logs from `make docs`

config_includer:
```
go generate -run="tools/config_includer/generator" ./plugins/inputs/...
2023/07/27 00:08:57 Reading template "\\plugins\\inputs\\http\\sample.conf.in" failed: open \plugins\inputs\http\sample.conf.in: The system cannot find the path specified.
plugins\inputs\http\http.go:1: running "../../../tools/config_includer/generator": exit status 1
```
readme_config_includer
```
2023/07/27 00:10:54 Ignoring include "C:\\docs\\includes\\plugin_config.md" as it cannot be found or is not a regular file...
```

### Additional info

This is happening because `filepath.IsAbs` is being used to check if a path starts with `/`. On windows however, this will return false for paths that start with slash. The following branch has a fix, but it's a bit messy in config_includer due to the paths sometimes having backslashes (the initial path), and sometimes having slashes (recursive includes), so there's probably some better cleanup that could be done.
https://github.com/influxdata/telegraf/compare/master...gazpachoking:telegraf:make_docs_windows

### PR DESCRIPTION
**Pull request by gazpachoking**
resolves #13681

`config_includer` and `config_readme_includer` currently don't work on Windows. This fixes that.

The underlying issue is that the tools test for 'absolute' includes, which are relative to the telegraf directory rather than the directory of the current file. Using `filepath.IsAbs` does not work for this test on Windows, and Go doesn't provide a way to do posix path tests/manipulation when running on Windows.

### CODE REVIEW

File: tools/config_includer/generator.go [hunk 1]
```diff
@@ -91,7 +91,7 @@ func main() {
 
 		newUnresolved := make(map[string]string)
 		for name, fn := range unresolved {
-			if filepath.IsAbs(fn) {
+			if strings.HasPrefix(fn, string(filepath.Separator)) || strings.HasPrefix(fn, "/") {
```
Comment (srebhan) [inline tools/config_includer/generator.go:94]: I think we only should allow slash based paths in the file... So
```suggestion
			if strings.HasPrefix(fn, "/") {
```
should be good enough.

File: tools/config_includer/generator.go [hunk 2]
```diff
@@ -121,7 +121,7 @@ func main() {
 			pwd = string(filepath.Separator) + pwd
 			for _, iname := range extractIncludes(tmpl) {
 				ifn := iname
-				if !filepath.IsAbs(ifn) {
+				if !strings.HasPrefix(ifn, "/") {
 					ifn = filepath.Join(pwd, ifn)
```
Comment (srebhan) [inline tools/config_includer/generator.go:125]: Your change is good but I wonder if the line below should read
```suggestion
					ifn = filepath.Join(pwd, filepath.FromSlash(ifn))
```

File: tools/readme_config_includer/generator.go [hunk 3]
```diff
@@ -55,7 +55,7 @@ func extractIncludeBlock(txt []byte, includesEx *regexp.Regexp, root string) *in
 		include := filepath.FromSlash(string(inc[1]))
 		// Make absolute paths relative to the include-root if any
 		// Check original value to avoid platform specific slashes
-		if filepath.IsAbs(string(inc[1])) {
+		if string(inc[1][0]) == "/" {
```
Comment (srebhan) [inline tools/readme_config_includer/generator.go:58]: Please also use
```suggestion
		if strings.HasPrefix(inc[1], "/") {
```

**Review by srebhan — COMMENTED** [review]
Thanks for the fix @gazpachoking! Some small comments...

File: tools/config_includer/generator.go (hunk 1 above)
Comment (gazpachoking) [reply tools/config_includer/generator.go:94]: I tried that. `filepath.Join` turns the slashes into backslashes on windows though, so we end up with a backslash from here. https://github.com/influxdata/telegraf/blob/185292dbc3b211605de9ee5e25f4dbbbe3860c1a/tools/config_includer/generator.go#L86

File: tools/config_includer/generator.go (hunk 2 above)
Comment (gazpachoking) [reply tools/config_includer/generator.go:125]: Windows doesn't mind slashes as path separators, but filepath.Join already turns them into backslashes anyway.

**Comment by gazpachoking** [comment]
My preference would have been to leave all the paths in posix format as much as possible, but Go doesn't provide any way to deal with posix paths on Windows. I think the latest solution is much cleaner than my first one though.

File: tools/config_includer/generator.go (hunk 1 above)
Comment (srebhan) [reply tools/config_includer/generator.go]: Understood. Can you please add this as a comment to the code so people will not "simplify" this as I was tempted to do... :-)

File: tools/config_includer/generator.go (hunk 2 above)
Comment (srebhan) [reply tools/config_includer/generator.go:125]: Cool, didn't know...

File: tools/config_includer/generator.go (hunk 1 above)
Comment (srebhan) [reply tools/config_includer/generator.go]: Oh saw you fixed it more elegantly... Thanks!

**Review by srebhan — APPROVED** [review]
Thanks for the clean fix @gazpachoking!

**Review by powersj — APPROVED** [review]

**Comment by powersj** [comment]
Thank you!

**Pull request merged**